skglm/solvers/cd_solver.py

```diff
@@ -276,15 +276,22 @@ def cd_solver(
         for epoch in range(max_epochs):
             if is_sparse:
                 _cd_epoch_sparse(
-                    X.data, X.indptr, X.indices, y, w, Xw, datafit, penalty,
-                    ws)
+                    X.data, X.indptr, X.indices, y, w[:n_features], Xw,
+                    datafit, penalty, ws)
             else:
-                _cd_epoch(X, y, w, Xw, datafit, penalty, ws)
+                _cd_epoch(X, y, w[:n_features], Xw, datafit, penalty, ws)
 
+            # update intercept
+            if fit_intercept:
+                intercept_old = w[-1]
+                w[-1] -= datafit.intercept_update_step(y, Xw)
+                Xw += (w[-1] - intercept_old)
             # 3) do Anderson acceleration on smaller problem
             # TODO optimize computation using ws
             if use_acc:
```

Comment:
I forgot, why don't we use `AndersonAccelerator` here ?

Reply:
because we still have to merge it: https://github.com/scikit-learn-contrib/skglm/pull/33